src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,212 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+var trackedControlsUtils = require('../utils/tracked-controls');
+
+// FIXME: need appropriate models, not the vive ones!
+var TOUCH_CONTROLLER_MODEL_OBJ_URL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.obj';
+var TOUCH_CONTROLLER_MODEL_OBJ_MTL = 'https://cdn.aframe.io/controllers/vive/vr_controller_vive.mtl';
+
+/**
+ * Oculus Touch Controls Component
+ * Interfaces with Oculus Touch controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('oculus-touch-controls', {
+  dependencies: ['tracked-controls'],
+
+  schema: {
+    hand: {default: 'left'},
+    buttonColor: { default: '#FAFAFA' },  // Off-white.
+    buttonHighlightColor: {default: '#22D1EE'},  // Light blue.
+    model: {default: true},
+    rotationOffset: {default: -999} // use -999 as sentinel value to auto-determine based on hand
+  },
+
+  idPrefix: 'Oculus Touch',
+
+  // buttonId
+  // 0 - thumbstick
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
+    'left': {
```

Comment:
FYI, because of the other comments on `menu` and `system` buttons (and moving the common `menu` button mapping) I needed to reinstate `oculus-touch.A-or-X` -- but didn't reinstate `oculus-touch.B-or-Y` yet since `menu` currently covers that.